### DISCUSSION TITLE
PHPDoc Annotation not changed to imported class

### DISCUSSION THREAD

**Discussion by dahaupt** [rectorphp/rector#7049] [Q&A] [answered]
I've noticed a different behavior after updating to v0.12.17.

Rector adds a new use statement:
```diff
+use TYPO3\CMS\Extbase\Annotation\Validate;
```

But does not change the related PHPDoc annotation:
```php
/**
* @TYPO3\CMS\Extbase\Annotation\Validate
*/
public function method(){}
```

PHP-CS-Fixer afterwards removes the unused import. 

I am not sure if this is specific to TYPO3 or a configuration mistake of myself. Thankful for any help!

**Comment by TomasVotruba** [comment, marked as answer]
Thanks for reporting. This looks like issue.
Could you share generic (non-typo3 project, but generic class) [demo link](http://getrector.org/demo)?

**Reply by dahaupt** [reply]
https://getrector.org/demo/13d25e4b-65c7-45bc-acf7-3bcf2e3dfa94

Does the demo page use the latest version? I get a different result locally using `v0.12.17` and the same configuration / test class.

**Reply by TomasVotruba** [reply]
Demo uses latest `dev-main` to test new feature before stable release.
One of differences, is that `TYPO3\CMS\Extbase\Annotation\ORM\Lazy` does not exist in the demo, as there is no definition for it.

It might be easier to isolate real code. Could you create minimal Github repository with reproducible? Thanks :+1:

**Reply by dahaupt** [reply]
Yes, I will do that asap.

**Reply by TomasVotruba** [reply]
Thanks!

If you can also create an issue instead of discussion, it would be great :+1:

**Reply by dahaupt** [reply]
Reproduction repository: https://github.com/dahaupt/rector-annotation-import
